lib/Cache.js

```diff
@@ -126,13 +126,46 @@ function Cache () {
   this.get = function (type, key, options, callback) {
     var self = this
     var table = type + ':' + key
-    this.db.select(table, options, function (err, result) {
+    var query = this.decodeGeoservices(options)
+    this.db.select(table, query, function (err, result) {
       if (err) return callback(err)
       self.process(type, key, result, options, callback)
     })
   }
 
   /**
+   * Translates a geoservices query into sql parts
+   *
+   * @param {object} query - an object with geoservices strings
+   * @return {object} parsed sql parts
+   */
+  this.decodeGeoservices = function (query) {
+    if (query.orderByFields && query.orderByFields.length) {
+      query.order_by = this._convertOrderByFields(query.orderByFields)
+      delete query.orderByFields
```

Comment:
I agree in principle. In this case, my specific goal is to translate the query object. You shouldn't actually think of the argument and return as the same object.